### PR TITLE
docs: Explicitly mention ODBC driver setup

### LINKED ISSUE #347: ODBC docs don't make it clear that you need drivers
If users don't wish to configure a docker image for their spicepod, the documentation doesn't make it clear that you need to install drivers to use the connector.

### PR DESCRIPTION
**Pull request by peasee**
## 🗣 Description

* Explicitly mention the requirement for ODBC drivers in the ODBC connector docs

## 🔨 Related Issues

* Closes #347

### CODE REVIEW

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 1]
```diff
@@ -4,6 +4,23 @@ sidebar_label: 'ODBC Data Connector'
...
 
+## Setup
+
+:::warning
+
+ODBC support is not included in the released binaries. To use ODBC with Spice, you need to [checkout and compile the code](https://github.com/spiceai/spiceai/blob/trunk/CONTRIBUTING.md#building) with the `--features odbc` flag (`cargo build --release --features odbc`).
+
+:::
+
+An ODBC connection requires a compatible ODBC driver. Whether running as a Docker image or directly on a computer, a valid ODBC driver configuration is necessary. ODBC drivers for various connection types are available from their respective vendors:
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/odbc.md:15]:
```suggestion
An ODBC connection requires a compatible ODBC driver. When run as a Docker image or on a computer, valid ODBC driver configuration is necessary. There are many ODBC drivers available from their respective vendors. For example:
```

Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
An ODBC connection requires a compatible ODBC driver and valid driver configuration. ODBC drivers are available from their respective vendors.  Here are a few examples:
```

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 2]
```diff
@@ -4,6 +4,23 @@ sidebar_label: 'ODBC Data Connector'
...
+
+- [PostgreSQL](https://odbc.postgresql.org/)
+- [MySQL](https://dev.mysql.com/downloads/connector/odbc/)
+- [Databricks](https://www.databricks.com/spark/odbc-drivers-download)
+- [AWS Athena](https://docs.aws.amazon.com/athena/latest/ug/connect-with-odbc.html)
+
+Please note that this is not an exhaustive list of drivers, as there are many ODBC adapters available.
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/odbc.md]: I think you can remove this sentence with the above suggestion.

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 3]
```diff
@@ -4,6 +4,21 @@ sidebar_label: 'ODBC Data Connector'
...
 
+## Setup
+
+:::warning
+
+ODBC support is not included in the released binaries. To use ODBC with Spice, you need to [checkout and compile the code](https://github.com/spiceai/spiceai/blob/trunk/CONTRIBUTING.md#building) with the `--features odbc` flag (`cargo build --release --features odbc`).
+
+:::
+
+An ODBC connection requires a compatible ODBC driver and valid driver configuration. ODBC drivers are available from their respective vendors. Here are a few examples:
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md:25]: It requires two components, a driver manager (i.e. `unixodbc`) and a driver to connect to the database. On Windows the driver manager is built-in, but Linux and macOS require `unixodbc` to be installed

**Review by phillipleblanc — DISMISSED** [review]

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 4]
```diff
@@ -19,6 +19,11 @@ An ODBC connection requires a compatible ODBC driver and valid driver configurat
 - [Databricks](https://www.databricks.com/spark/odbc-drivers-download)
 - [AWS Athena](https://docs.aws.amazon.com/athena/latest/ug/connect-with-odbc.html)
 
+Unix additionally requires the installation of `unixodbc`. Here are some examples for installing this package:
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/odbc.md]:
```suggestion
Non-Windows systems additionally require the installation of an ODBC Driver Manager like `unixodbc`.
```

File: spiceaidocs/docs/components/data-connectors/odbc.md [hunk 5]
```diff
@@ -4,6 +4,26 @@ sidebar_label: 'ODBC Data Connector'
 description: 'ODBC Data Connector Documentation'
 ---
 
+## Setup
+
+:::warning
+
+ODBC support is not included in the released binaries. To use ODBC with Spice, you need to [checkout and compile the code](https://github.com/spiceai/spiceai/blob/trunk/CONTRIBUTING.md#building) with the `--features odbc` flag (`cargo build --release --features odbc`).
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/odbc.md:11]: Or use the official Docker image - see language from release notes: https://github.com/spiceai/spiceai/pull/2132/files

File: spiceaidocs/docs/components/data-connectors/odbc.md (hunk 4 above)
Comment (lukekim) [reply spiceaidocs/docs/components/data-connectors/odbc.md:22]: "In addition, ..."

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**